### ISSUE TITLE
does this work with static ips from att?

### ISSUE THREAD

**Issue by nikolaishields** [jaysoffian/eap_proxy#39]
I know there was a [prior closed issue ](https://github.com/jaysoffian/eap_proxy/issues/18 )that seems to have some resolution or manner for utilizing static IP's provided by ATT.
Any help or guidance on this issue would be greatly appreciated, I'd also be more than happy to document the process and open a pr to add to the README/project.

**Comment by bvdrax** [comment]
I found various instructions over the years, but never got any of them to work successfully. I would be super interested in this.

**Comment by jaysoffian** [comment]
I can't think of any reason why this would not work with a static IP block, but I can't help debug the issue. You're going to have to use tcpdump and/or the `--debug` and `--debug-packets` switch to `eap_proxy` to figure out what's going on. Sorry I can't help any more than that.

**Comment by ramsaso** [comment]
Hi, I was the person that got the static ips to work for me.

I think these were the steps I took to get ipv6 to work. 
The equipment you will need is a smart switch or a switch capable of VLANs.

(Note that it is important to know what ip blocks you were assigned to by AT&T and what your netmask is. For example, x.x.x.x/27 denotes 32 ip addresses but only 29 usable ones)

1. Make a VLAN in the Dashboard for your LAN interface (like for example, eth2 (in this case, my LAN) but with a vlan id of 100, it would be eth2.100)

2. Assign there, the last ip that you have in your block that is NOT A BROADCAST IP ADRRESS and suffixing it with the netmask.

3. In Firewall/NAT and selecting the NAT submenu, add a Source and Destination NAT rules ( 2 rules in total, one for Source and one for Destination)

3.1. For the Source NAT rule, have it be excluded from NAT and make the outbound Interface to be the WAN port for which the public ip of the former RG is (Ex. WAN VLAN 0, which is the public ip address pulled from DHCP or in my case, eth0.0)

- Be sure to choose "Use Masquerade" in Translation. (It is vital that this is selected, lest you have no connectivity at all for your statics)

- Make sure 'Protocol' is set to All protocols.
- Source Address (Src Address) is going to be the IP address that you just used, when you created your VLAN.
- Save, make that the top rule order (clicking and dragging it to the top) and save it.
- I do not know if this is important enough to warrant turning on, but have Enable Logging checkmarked..

3.2. For Destination NAT, it's mostly the same thing as the Source NAT rule you just did.
HOWEVER.

- Make the Destination Address (Dest Address) to the ip address you put in the dashboard for the VLAN you have created but don't forget the netmask.

- Translations is going to be like above but sans the netmask (without the netmask included).

- Save, make it the top rule order and save it.

Voila, you should be able to get your static ips to work in that manner by manually assigning an ip address to a device you have.

On your VLAN capable switch, this is what I did to get my static ips to work.
(This is based off of a smart switch by TP-Link, the TL-SG108E, that I own). 
(Because of this, your configuration may vary.)

---- MAKE SURE THAT THE VLAN CAPABLE SWTICH IS CONNECTED DIRECTLY TO THE LAN INTERFACE OF YOUR EDGEROUTER----
1. In your VLAN settings, make sure you are in 802.1Q VLAN
2. On VLAN ID, type in the VLAN ID that you are using for your static ip blocks (for me, it would be 100.)

(THIS IS VERY CRITICAL. 
- Untagged means that the traffic for only the VLAN ID you set a port to, only flows in that route.
- Tagged means that all traffic is passed through the port, regardless of VLAN ID it is set to, but only traffic destined to a device with a set VLAN ID is noted with a special 'tag' for the device to pick up, in layman terms. {Please correct me if I am wrong with my theory}

3. If you are going to attach another switch to the port and have that connected to both unaltered and altered VLAN devices, select 'Tagged'. 

4. If you are going to connect a device for which you will NOT define a VLAN ID in software, select 'Untagged'.

5. Select (for the port with the altered VLAN ID) and make sure the PVID is set to the ID you chose in the edgerouter. (this does not apply to tagged ports, at least for me)

6. Save.

You may also make a DHCP server by following this example.

1. Range Start is the beginning of your ip block
2. Range Stop is the end of your ip block (THIS DOES NOT INCLUDE THE IP ADDRESS USED FOR MAKING THE VLAN)
3. Router is to be the IP address in the VLAN.
4. DNS 1 and 2 to be the DNS servers of your choice.
5. Save.
